plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/launcher/DataflowPipelineLaunchDelegate.java

```diff
@@ -114,25 +130,80 @@ public void launch(
           "Could not retrieve Pipeline Options Hierarchy for project " + projectName, e));
     }
 
-    if (!pipelineConfig.isValid(hierarchy, getPreferences(pipelineConfig))) {
+    DataflowPreferences preferences = ProjectOrWorkspaceDataflowPreferences.forProject(project);
+    if (!pipelineConfig.isValid(hierarchy, preferences)) {
       throw new IllegalArgumentException(
           "Provided Dataflow Pipeline Configuration is not valid: " + pipelineConfig.toString());
     }
 
-    List<String> argComponents = getArguments(configuration, pipelineConfig, hierarchy);
+    Map<String, String> effectiveArguments = getEffectiveArguments(pipelineConfig, preferences);
+
+    List<String> argComponents =
+        getArguments(configuration, pipelineConfig, hierarchy, effectiveArguments);
 
     ILaunchConfigurationWorkingCopy workingCopy =
         configuration.copy(DATAFLOW_LAUNCH_CONFIG_WORKING_COPY_PREFIX + configuration.getName());
     workingCopy.setAttribute(
         IJavaLaunchConfigurationConstants.ATTR_PROGRAM_ARGUMENTS, SPACE_JOINER.join(argComponents));
 
+    String accountEmail = effectiveArguments.get("accountEmail");
+    setLoginCredential(workingCopy, accountEmail);
+
     delegate.launch(workingCopy, mode, launch, progress.newChild(1));
   }
 
+  @VisibleForTesting
+  void setLoginCredential(ILaunchConfigurationWorkingCopy workingCopy, String accountEmail)
+      throws CoreException {
+    Preconditions.checkNotNull(accountEmail, "account email not set in the launch configuration");
+
+    try {
+      if (accountEmail.isEmpty()) {
+        String message = "No Google account selected for this launch.";
+        throw new CoreException(new Status(Status.ERROR, DataflowCorePlugin.PLUGIN_ID, message));
+      }
+
+      Credential credential = loginService.getCredential(accountEmail);
+      if (credential == null) {
+        String message = "The Google account saved for this lanuch is not logged in.";
+        throw new CoreException(new Status(Status.ERROR, DataflowCorePlugin.PLUGIN_ID, message));
+      }
+
+      // Dataflow SDK doesn't yet support reading credentials from an arbitrary JSON, so we use the
+      // workaround of setting the "GOOGLE_APPLICATION_CREDENTIALS" environment variable.
+      Map<String, String> variableMap = workingCopy.getAttribute(
+          ILaunchManager.ATTR_ENVIRONMENT_VARIABLES, new HashMap<String, String>());
+      if (variableMap.containsKey(GOOGLE_APPLICATION_CREDENTIALS_ENVIRONMENT_VARIABLE)) {
+        String message = "You cannot define the environment variable GOOGLE_APPLICATION_CREDENTIALS"
+            + " when launching Dataflow pipelines from Cloud Tools for Eclipse.";
+        throw new CoreException(new Status(Status.ERROR, DataflowCorePlugin.PLUGIN_ID, message));
+      }
+
+      Path jsonCredential = Files.createTempFile("google-ct4e-" + workingCopy.getName(), ".json");
```

Comment:
Will this work if there are multiple accounts logged in, and the suer is deploying from each of them?